linux/system/timezone.sls

```diff
@@ -3,9 +3,14 @@
 
 {%- if system.timezone is defined %}
 
+include:
+- linux.system.rsyslog
+
 {{ system.timezone }}:
   timezone.system:
   - utc: {{ system.utc }}
+  - watch_in:
+    - service: rsyslog_service
```

Comment:
I wouldn't deal with this: if you're changing the TZ setting, there are many other services that need to be restarted. In practice, you're better off to reboot the node. Also the heka.log_collector state should be reapplied.